feat: Introducing telephony module

leela opened the pull request:
Introducing a module named Telephony, that holds all the common doctype's needed to integrate any telephony communication app.

**Added**
* New module named Telephony added.
* `Incoming Call Settings` and `Incoming Call Handling Schedule` doctype's added
* Moved call_log doctype from communication into Telephony module
* Added validations for new doctype's
* Phone icon that is part of frappe is moved into ERP

**Screenshot**
![incoming_call_settings](https://user-images.githubusercontent.com/36557/100583256-73c95600-3310-11eb-9481-255839ca4764.png)

> no-docs

Docs will be added once we are done with one of the telephony integrations.

leela committed:
feat: Incoming call settings doctype added

Telephony module is created and `Incoming call settings`, `Incoming call
handling schedule` doctypes are added into it.

leela committed:
Incoming call settings validations

leela committed:
move call_log doctype into telephony module

leela committed:
Move phone icon code from frappe to telephony

leela committed:
Type annotations for incoming call dt validations

leela committed:
code formatting  according to standards

surajshetty3416 commented on erpnext/telephony/doctype/incoming_call_settings/incoming_call_settings.js:
```diff
@@ -0,0 +1,102 @@
+	let group_by_day = groupby(schedule, 'day_of_week');
+	for (const [day, day_schedule] of Object.entries(group_by_day)) {
+		if (is_call_schedule_overlapped(day_schedule)) {
+			frappe.throw(`Please fix overlapping time slots for ${day}.`);
```
```suggestion
			frappe.throw(__("Please fix overlapping time slots for {0}.", [day]);
```
User facing strings should be translatable.
https://github.com/frappe/erpnext/wiki/Coding-Standards#user-facing-messages-must-be-translated

More information on translations: https://frappeframework.com/docs/user/en/translations

surajshetty3416 commented on erpnext/telephony/doctype/incoming_call_settings/incoming_call_settings.js:
```diff
@@ -0,0 +1,102 @@
+		let from_time_in_secs = time_to_seconds(record.from_time);
+		let to_time_in_secs = time_to_seconds(record.to_time);
+		if (from_time_in_secs >= to_time_in_secs) {
+			errors.push(`Call Schedule Row ${row}: To time slot should always be ahead of From time slot.`);
```
```suggestion
			errors.push(__("Call Schedule Row {0}: To time slot should always be ahead of From time slot."), [row]);
```

surajshetty3416 commented on erpnext/telephony/doctype/incoming_call_settings/incoming_call_settings.py:
```diff
@@ -0,0 +1,62 @@
+			from_time = self.time_to_seconds(record.from_time)
+			to_time = self.time_to_seconds(record.to_time)
+			if from_time >= to_time:
+				errors.append(
+					f'Call Schedule Row {record.idx}: To time slot should always be ahead of From time slot.'
+				)
```
```suggestion
				errors.append(_("Call Schedule Row {0}: To time slot should always be ahead of From time slot.").format(record.idx))
```
https://github.com/frappe/erpnext/wiki/Coding-Standards#user-facing-messages-must-be-translated

surajshetty3416 commented on erpnext/telephony/doctype/incoming_call_settings/incoming_call_settings.py:
```diff
@@ -0,0 +1,62 @@
+
+			for i in range(1, len(timeslots)):
+				if self.check_timeslots_overlap(timeslots[i-1], timeslots[i]):
+					frappe.throw(f"Please fix overlapping time slots for {day}.")
```
```suggestion
					frappe.throw(_("Please fix overlapping time slots for {0}.").format(day))
```

leela committed:
changing user facing strings into translatable

leela committed:
indent code to 4 columns instead of 8 in github

Github respect editorconfig settings. Adding indent_size as 4 in
settings makes github indent code to 4 columns.

leela committed:
add missing frappe._ import

ameerbaathar commented:
how to use the API for this feature ?

nashwan762 commented:
O
